bazel/envoy_build_system.bzl

```diff
@@ -455,15 +455,17 @@ def envoy_cc_test(
     test_lib_tags = []
     if coverage:
         test_lib_tags.append("coverage_test_lib")
-    envoy_cc_test_library(
-        name = name + "_lib",
+    _envoy_cc_test_infrastructure_library(
+        name = name + "_lib_INTERNAL_ONLY",
         srcs = srcs,
         data = data,
         external_deps = external_deps,
-        deps = deps,
+        deps = deps + [repository + "//test/test_common:printers_includes"],
         repository = repository,
         tags = test_lib_tags,
         copts = copts,
+        # Restrict only to the code coverage tools.
+        visibility = ["@envoy//test/coverage:__pkg__"],
```

Comment:
Can you verify that the fuzzers targets needed at https://github.com/google/oss-fuzz/blob/master/projects/envoy/build.sh#L62 still build? I.e. the `_driverless` variants of each fuzz target. This isn't caught by CI.

Reply:
Done. Verified that `bazel build $(bazel query 'filter("_driverless", kind(".* rule", //...))')` progresses to the linking stage (despite encountering a library error there) and that  `bazel build $(bazel query 'filter("_corpus$", kind(".* rule", //...))')` finishes successfully.